src/NHibernate/Driver/DriverBase.cs

```diff
@@ -329,6 +329,55 @@ public virtual void AdjustCommand(DbCommand command)
 		{
 		}
 
+#if NET6_0_OR_GREATER
+		public void PrepareBatch(DbBatch batch)
+		{
+			OnBeforePrepare(batch);
+
+			if (SupportsPreparingCommands && prepareSql)
+			{
+				batch.Prepare();
+			}
+		}
+
+		/// <summary>
+		/// Override to make any adjustments to the DbBatch object.  (e.g., Oracle custom OUT parameter)
+		/// Parameters have been bound by this point, so their order can be adjusted too.
+		/// This is analogous to the RegisterResultSetOutParameter() function in Hibernate.
+		/// </summary>
+		protected virtual void OnBeforePrepare(DbBatch batch)
+		{
+		}
+
+		public virtual DbBatch CreateBatch()
+		{
+			throw new NotSupportedException();
```

Comment:
Switch to not supported (was not implemented) for consistency with the interface default implementation, en because ["not implemented" means "still in dev"](https://learn.microsoft.com/en-us/dotnet/fundamentals/runtime-libraries/system-notimplementedexception#throw-the-exception) while ["not supported"](https://learn.microsoft.com/en-us/dotnet/fundamentals/runtime-libraries/system-notsupportedexception#throw-a-notsupportedexception-exception) is meant for implementations that might not support some features of a base class / interface. When `CanCreateBatch` yields `false`, it is adequate to throw `NotSupportedException` from `CreateBatch`.